external-providers/generic-external-provider/pkg/generic/service_client.go

```diff
@@ -144,85 +180,144 @@ func parallelWalk(location string, regex *regexp.Regexp) ([]protocol.TextDocumen
 	return positions, nil
 }
 
-func (p *genericServiceClient) GetAllSymbols(ctx context.Context, query string) []protocol.WorkspaceSymbol {
+// Returns all symbols for the given query.
+// NOTE: Only returns definitions when server does not supoprt workspace/symbol.
+// Is is intended behavior?
```

Comment:
For the .NET work, what made the most sense while experimenting was to:
1. find occurrences of pattern
2. call "textDocument/definition" and see if the response URI includes the referenced namespace (ie. System.Web.Mvc)...if it does then we add it as an incident.

And, iirc @fabianvf had mentioned gopls handling `references` differently which would mean that this is yet another area where differences in implementation may bite us. Makes me wonder if we want to make it possible to pass a `callback` of sorts that allows  you to specify a function for determining if a found Symbol using the fallback method should be included as an incident.

Reply:
If I'm understanding right, you're saying the .NET work utilizes the fallback method? I added the note as if we use the fallback method, only the definitions will be returned, rather than every symbol like when using `workspace/symbol`. Honestly, I'm not quite seeing why we should do anything other than the fallback method?

Reply:
I copy/pasted and then modified the fallback method for what I was doing with the .NET work. See [here](https://github.com/konveyor/analyzer-dotnet-provider/pull/7/files#diff-908ae641af05346d94599b805eeb64ded6e475d6f461d8a06e09facdb4443291R81).

My thought was that it may be possible for us to create an exported method for our fallback that accepts a `func` for determining if a given symbol is the one the provider is looking for.

Reply:
Looking more thoroughly at what I did, it may be best for us to have our fallback method as an exported function that simply returns a slice of positions...leaving it up to the caller on how to sort which of those positions are valid symbols or not.